### PR TITLE
HPCC-15591 DESDL - use loaded service name as actual service name

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review.

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 1]
```diff
@@ -139,15 +139,21 @@ IPropertyTree * fetchESDLBindingFromDali(const char *process, const char *bindin
...
 IPropertyTree * fetchESDLBindingFromStateFile(const char *process, const char *bindingName, const char * stateFileName)
 {
-    Owned<IPTree> esdlState = createPTreeFromXMLFile(stateFileName);
-    if (!esdlState)
-        throw MakeStringException(-1, "Could not load DESDL binding state from local file: '%s' for ESP binding %s.%s", stateFileName, process, bindingName);
+    try
+    {
+        Owned<IPTree> esdlState = createPTreeFromXMLFile(stateFileName);
+        if (!esdlState)
+            throw MakeStringException(-1, "Could not load DESDL binding state from local file: '%s' for ESP binding %s.%s", stateFileName, process, bindingName);
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: If you have the line 154, why do you need to throw the exception here?

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 2]
```diff
@@ -987,20 +998,20 @@ EsdlBindingImpl::EsdlBindingImpl(IPropertyTree* cfg, const char *binding,  const
...
+        //Subscribe to Dali anyway
+		DBGLOG("ESDL Binding %s is subscribing to all /ESDL/Bindings/Binding dali changes", binding);
+		//Since it seems we cannot subscribe to a specific /ESDL/Bindings/Binding, all bindings have to subscribe to the /ESDL/Bindings branch
+		m_pBindingSubscription.clear();
+		m_pBindingSubscription.setown( new CESDLBindingSubscription(this));
+
+		DBGLOG("ESDL Binding %s is subscribing to all /ESDL/Bindings/Definition dali changes", binding);
+		//Since it seems we cannot subscribe to a specific /ESDL/Definitions/Definition, all bindings have to subscribe to the /ESDL/Definitions branch
+		m_pDefinitionSubscription.clear();
+		m_pDefinitionSubscription.setown( new CESDLDefinitionSubscription(this));
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: minor: incorrect indent.

**Comment by wangkx** [comment]
@rpastrana back to you.

**Comment by rpastrana** [comment]
@wangkx please review latest changes.

**Comment by wangkx** [comment]
@rpastrana looks good now.

**Comment by rpastrana** [comment]
@richardkchapman please merge

**Pull request merged**